### PR TITLE
1.0.0-b32

### PR DESCRIPTION
**Pull request by vinniefalco**
* Add io_service completion invariants tests 

API Changes:
* Remove http Writer suspend and resume feature (API Change):  …

Documentation preview:
http://vinniefalco.github.io/stage/beast/b32/

### CODE REVIEW

**Comment by vinniefalco** [comment]
Why does it look like coverage went down?

**Comment by hoditohod** [comment]
:+1:  Looks good to me!

File: doc/types/Writer.qbk [hunk 1]
```diff
@@ -167,17 +147,12 @@ public:
...
-                `false` when there may be more data,
-                boost::indeterminate to suspend.
-
-        @note Undefined behavior if the callee takes ownership
-              of resume but does not return boost::indeterminate.
+        @return `true` if there is no more data to send,
+                `false` when there may be more data.
     */
     template<class WriteFunction>
     boost::tribool
```
Comment (nbougalis) [inline doc/types/Writer.qbk]: Shouldn't this be changed to `bool`?

File: CMakeLists.txt [hunk 2]
```diff
@@ -19,24 +20,27 @@ if (MSVC)
...
-    link_directories(${Boost_LIBRARY_DIR})
-
     set(THREADS_PREFER_PTHREAD_FLAG ON)
     find_package(Threads)
 
     set(CMAKE_CXX_FLAGS
       "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra -Wpedantic -Wno-unused-parameter -DBOOST_COROUTINES_NO_DEPRECATION_WARNING=1")
 endif()
 
+set(Boost_USE_STATIC_LIBS ON)
```
Comment (ras0219-msft) [inline CMakeLists.txt]: This will override the `option()` setting above.

File: doc/types/Writer.qbk (hunk 1 above)
Comment (vinniefalco) [inline doc/types/Writer.qbk]: Will fix

File: CMakeLists.txt (hunk 2 above)
Comment (vinniefalco) [inline CMakeLists.txt]: Should I just delete the line?

**Comment by vinniefalco** [comment]
I pushed two commits to address the review comments

File: CMakeLists.txt (hunk 2 above)
Comment (ras0219-msft) [inline CMakeLists.txt]: Yes, I think that's the best option.

Comment (vinniefalco) [inline CMakeLists.txt]: Will fix

**Pull request merged**